synapse/storage/databases/main/receipts.py

```diff
@@ -702,9 +702,6 @@ def _insert_linearized_receipt_txn(
                 "data": json_encoder.encode(data),
             },
             where_clause=where_clause,
-            # receipts_linearized has a unique constraint on
-            # (user_id, room_id, receipt_type), so no need to lock
-            lock=False,
```

Comment:
To double check: is the table correctly deemed safe to upsert into when the relevant background updates have run? (Wasn't sure how the second commit would affect this, if at all)

Reply:
Yes, once the unique index has been added by the background update, we will be able to rely on native upserts again (and the value of `lock` won't matter).